modules/ROOT/pages/salesforce/salesforce-connector-97.adoc

```diff
@@ -0,0 +1,1274 @@
+
+If the *Disable session invalidation* field is set to False, the connector automatically destroys the session after it's no longer needed.
+
+You should keep the session alive when you are working with threads or concurrency in general. Salesforce uses
+the same session for all your threads (for example, if you have an active session and you log in again, Salesforce uses the existing session instead of creating a new one). To make sure the connection doesn't close when a thread is finished, you should set the *Disable session invalidation* field to True in the Connection section of the connector's global element properties.
+
+image::salesforce/salesforce-disable-session.png[Disable Session Field]
+
+=== Use Mutual TLS
+
+In v9.7.0 and later, all authentication types support Mutual TLS. To use this you need a keystore file in the JKS format, and a password for it. See <<Generate a Keystore File>> for more information.
+
+Simply specify the path to the keystore file and the password in the configuration window (as shown in the image below) and any user that requires Mutual TLS authentication is able to login using the connector.
+
+image::salesforce/salesforce-mutual-tls.png[Mutual TLS]
+
+To set up the Mutual TLS certificate in your Salesforce environment, see https://help.salesforce.com/articleView?id=security_keys_uploading_mutual_auth_cert.htm&type=5[Set Up a Mutual Authentication Certificate].
+
+=== Configure Apex Settings
+
+You can set Apex REST and SOAP access using Studio or in XML. When you connect to Salesforce, the Salesforce connector gets the names of the Apex classes and methods belonging to them that can be invoked.
+
+All Salesforce connection configurations support these Apex settings:
+
+* *Fetch All Apex SOAP Metadata*- Fetches the metadata of all the Apex SOAP classes. Takes precedence over Apex Class Name settings. 
+* *Fetch All Apex REST Metadata* - Fetches the metadata of all the all Apex REST classes. Takes precedence over Apex Class Name settings.
+* *Apex Class Names* - List of Apex class names to use for limiting the set of classes you fetch along with the methods they expose. This setting can speed the fetch process if there are a lot of classes that you do not need to fetch.
+
+You can provide Apex settings in Design Center and in Anypoint Studio 7.
+
+Click the *Apex* tab to choose the settings.
+
+Apex settings values:
+
+* *Fetch All Apex SOAP Metadata* - Fetches the metadata of all the Apex SOAP classes.
+* *Fetch All Apex REST Metadata* - Fetches the metadata of all the all Apex REST classes.
+
+Apex Class Names:
+
+The *Expression*, *Edit inline*, or *Bean reference* choices provide these options:
+
+* *None* - No Apex class name is mentioned for DataSense to acquire.
+* *From a message* - Lets you specify the class name using an expression.
+* *Create object manually* - You can create a list and add class names to the list - only those classes and their methods are acquired by DataSense.

```

Comment:
The late part o the sentence is confusing,

Reply:
Changed to: 
* *Create object manually* - DataSense acquires only those classes and methods that you put in a list.